clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -204,16 +204,46 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
       : null
 
     def migCreateOperation
+    def instanceGroupManager = new InstanceGroupManager()
+        .setName(serverGroupName)
+        .setBaseInstanceName(serverGroupName)
+        .setInstanceTemplate(instanceTemplateUrl)
+        .setTargetSize(description.targetSize)
+        .setTargetPools(targetPools)
+        .setAutoHealingPolicies(autoHealingPolicy)
+
+    def hasBackendServices = description.instanceMetadata &&
+        description.instanceMetadata.containsKey(GoogleServerGroup.View.BACKEND_SERVICE_NAMES)
+
+    if (hasBackendServices && (description?.loadBalancingPolicy || description.source.serverGroupName))  {
+      NamedPort namedPort = null
+      def sourceManagerName = description.source.serverGroupName
+      if (sourceManagerName) {
+        def sourceInstanceGroupManager
+        if (isRegional) {
+          sourceInstanceGroupManager = compute.regionInstanceGroupManagers().get(project, description.source.region, sourceManagerName).execute()
```

Comment:
I think a better approach might be to cache the named ports on server groups in the caching agents, and then retrieve the source server group from the cache here instead of making live calls.